### PR TITLE
Get contracts and test suite working with Optimistic Ethereum

### LINKED ISSUE #8: Update to run on OVM
Via Kevin:

> you should be able to do a truffle config like this: https://github.com/ethereum-optimism/optimism-monorepo/blob/d117cf2e7810db5b85f1fea672c75fb74c57a249/packages/ovm-toolchain/test/config/truffle-config.js#L1-L38

> and replace the import ganache from ovm-toolchain with import * as ganache from @eth-optimism/plugins/ganache see https://github.com/ethereum-optimism/plugins

> we have deprecated the ovm-toolchain in favor of @eth-optimism/plugins

### PR DESCRIPTION
**Pull request by apbendi**
* Downgrade to Solidity 0.5.16
* Create an OVM specific truffl config and test
  runner script
* Skip migrations when testing on the OVM
* Update tests to conditionally skip assertions and
  other tests that are not applicable on the OVM
* Update README

closes #8

### CODE REVIEW

File: contracts/MultiSigWallet.sol [hunk 1]
```diff
@@ -244,11 +244,12 @@ contract MultiSigWallet {
     // of the Solidity's code generator to produce a loop that copies tx.data into memory.
     function external_call(address destination, uint value, uint dataLength, bytes memory data) internal returns (bool) {
         bool result;
+
         assembly {
             let x := mload(0x40)   // "Allocate" memory for output (0x40 is where "free memory" pointer is stored by convention)
             let d := add(data, 32) // First 32 bytes are the padded length of data, so exclude that
             result := call(
-                sub(gas, 34710),   // 34710 is the value that solidity is currently emitting
+                sub(gas, 0),   // 34710 is the value that solidity is currently emitting
```
Comment (wildmolasses) [inline contracts/MultiSigWallet.sol]: Might recommend changing the comment here since we're no longer using value of 34710. Personally could use a note on why this value changed. i'm also curious on why this change doesn't break any EVM tests.

**Review by wildmolasses — CHANGES_REQUESTED** [review]
Looks great, just 1 change requested.

File: contracts/MultiSigWallet.sol (hunk 1 above)
Comment (apbendi) [reply contracts/MultiSigWallet.sol]: Good call, reverted to original number and opened #15 to investigate further
